src/PackageUrl.cs

```diff
@@ -136,6 +141,7 @@ public override string ToString()
             if (Namespace != null)
             {
                 purl.Append(WebUtility.UrlEncode(Namespace));
+                purl.Replace(EncodedSlash, "/");
```

Comment:
I admit this seems a bit odd to me (particularly since it applies to all package types, not just `golang`), but since the parse method below collects any/all extra `/` characters into the namespace component, I think it should be fine.